engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -167,7 +168,7 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 		this.messageInterpolator = messageInterpolator;
 		this.traversableResolver = traversableResolver;
 		this.beanMetaDataManager = beanMetaDataManager;
-		this.parameterNameProvider = parameterNameProvider;
+		this.parameterNameProvider = new ExecutableParameterNameProvider( parameterNameProvider );
```

Comment:
Shouldn't we wrap it once and for all in `ValidatorFactoryImpl` instead of doing it this way? I'm especially worried about the fact that we might inadvertently wrap it several times.